src/Storage/UserGenerator/CsvFilesystemStorage.php

```diff
@@ -94,7 +94,7 @@ protected function castAssignments(AssignmentCollection $assignments): array
         $res = [];
         foreach ($assignments as $assignment) {
             /** @var Assignment $assignment */
-            $res[] = [$assignment->getUser()->getUsername(), $assignment->getLineItem()->getSlug()];
+            $res[] = [(string)$assignment->getUser()->getUsername(), $assignment->getLineItem()->getSlug()];
```

Comment:
if we have an opportunity to get null here. it should be checked before and skip not valid value or throw an exception